Task2/Kuba/NoughtsAndCrosses/GameBoard.cs

```diff
@@ -0,0 +1,129 @@
+        {
+            gameBoard = new GameBoardStatus[boardSize, boardSize];
+        }
+
+        public void MarkPlayerTurn(Player player, Position coordinates)
+        {
+            if (CoordinatesAreValid(coordinates))
+            {
+                this.gameBoard[coordinates.Vertical, coordinates.Horizontal] = player.ChoosenSymbol;
+            }
+        }
+
+        public bool BoardIsFull()
+        {
+            for (int i = 0; i < boardSize; i++)
+            {
+                for (int j = 0; j < boardSize; j++)
+                {
+                    if (this.gameBoard[i, j] == GameBoardStatus.empty) return false;
+                }
+            }
+            return true;
+        }
+
+        public bool PlayerWon(Player player)
+        {
+            return (CheckVertically(player) || CheckHorizontally(player) || CheckDiagonally(player));
+        }
+
+        private bool CheckVertically(Player player)
+        {
+            for (int i = 0; i < boardSize; i++)
+            {
+                int playerSymbolCount = 0;
+                for (int j = 0; j < boardSize; j++)
+                {
+                    if (this.gameBoard[j, i] == player.ChoosenSymbol)
+                    {
+                        playerSymbolCount++;
+                    }
+                }
+                if (playerSymbolCount == boardSize)
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        private bool CheckHorizontally(Player player)
+        {
+            for (int i = 0; i < boardSize; i++)
+            {
+                int playerSymbolCount = 0;
+                for (int j = 0; j < boardSize; j++)
+                {
+                    if (this.gameBoard[i, j] == player.ChoosenSymbol)
+                    {
+                        playerSymbolCount++;
+                    }
+                }
+                if (playerSymbolCount == boardSize)
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        private bool CheckDiagonally(Player player)
+        {
+            int playerSymbolCount = 0;
+            for (int i = 0; i < boardSize; i++)
+            {
+                if (this.gameBoard[i, i] == player.ChoosenSymbol)
+                {
+                    playerSymbolCount++;
+                }
+            }
+            if (playerSymbolCount == boardSize)
+            {
+                return true;
+            }
+            else
+            {
+                playerSymbolCount = 0;
+            }
+
+            for (int i = 0; i < boardSize; i++)
+            {
+                if (this.gameBoard[i, (boardSize - 1) - i] == player.ChoosenSymbol)
+                {
+                    playerSymbolCount++;
+                }
+            }
+            if (playerSymbolCount == boardSize)
+            {
+                return true;
+            }
```

Comment:
I think these two sections (for two diagonals) could be refactored to avoid code duplication.